tiledb/sm/storage_manager/context.h

```diff
@@ -117,6 +117,10 @@ class Context {
     return &(resources_.io_tp());
   }
 
+  [[nodiscard]] inline shared_ptr<RestClient> rest_client() const {
```

Comment:
It would be better to have this return `RestClient&` and throw on error. 

Among other benefits, it would get rid of a large amount of error handling copypasta in the four call sites in `tiledb.cc` where it appears. Eventually, it will allow more informative error handling overall, since we'll be able to distinguish "failed because facility not compiled in" vs. "failed because of some runtime error".

For the time being we can implement with `*(x.get())`. Later on we could mirror the `shared_ptr` with a reference wrapper or perhaps a some other mechanism that depends upon the compile configuration.